### PR TITLE
[v5] Use higher level functions in key validation tests

### PR DESCRIPTION
**Pull request by larabr**
- Use `key.keyPacket.validate` instead of `crypto.publicKey.validateParams`, see https://github.com/openpgpjs/openpgpjs/pull/1116#discussion_r447781386

- `key.decrypt` now only throws on error, no other value is returned

- Fix typo (rebase error) that caused tests to fail in Safari for p521

### CODE REVIEW

File: test/general/brainpool.js [hunk 1]
```diff
@@ -184,7 +184,7 @@ EJ4QcD/oQ6x1M/8X/iKQCtxZP8RnlrbH7ExkNON5s5g=
     const pk = await openpgp.key.readArmored(data[name].priv);
     expect(pk).to.exist;
     expect(pk.getKeyId().toHex()).to.equal(data[name].id);
-    expect(await pk.decrypt(data[name].pass)).to.be.true;
+    await expect(pk.decrypt(data[name].pass)).to.not.be.rejected;
```
Comment (twiss) [inline test/general/brainpool.js:187]: I would just write
```suggestion
    await pk.decrypt(data[name].pass);
```
(and similarly elsewhere); if it throws, the test will fail also.

File: test/general/key.js [hunk 2]
```diff
@@ -2203,13 +2203,13 @@ function versionSpecificTests() {
     const armor1 = key.armor();
     const armor2 = key.armor();
     expect(armor1).to.equal(armor2);
-    expect(await key.decrypt('passphrase')).to.be.true;
+    await expect(key.decrypt('passphrase')).to.not.be.rejected;
```
Comment (twiss) [inline test/general/key.js:2206]:
```suggestion
    await key.decrypt('passphrase');
```

File: test/general/key.js [hunk 3]
```diff
@@ -2203,13 +2203,13 @@ function versionSpecificTests() {
...
     expect(key.isDecrypted()).to.be.true;
     await key.encrypt('new_passphrase');
     expect(key.isDecrypted()).to.be.false;
-    await expect(key.decrypt('passphrase')).to.eventually.be.rejectedWith('Incorrect key passphrase');
+    await expect(key.decrypt('passphrase')).to.be.rejectedWith('Incorrect key passphrase');
     expect(key.isDecrypted()).to.be.false;
-    expect(await key.decrypt('new_passphrase')).to.be.true;
+    await expect(key.decrypt('new_passphrase')).to.not.be.rejected;
```
Comment (twiss) [inline test/general/key.js:2212]:
```suggestion
    await key.decrypt('new_passphrase');
```

File: test/general/openpgp.js [hunk 4]
```diff
@@ -374,6 +374,72 @@ EQr2Mx42THr260IFYp5E/rIA
 =oA0b
 -----END PGP PRIVATE KEY BLOCK-----`;
 
+
```
Comment (twiss) [inline test/general/openpgp.js:377]:
```suggestion
```

File: test/general/openpgp.js [hunk 5]
```diff
@@ -640,8 +709,8 @@ module.exports = () => describe('OpenPGP.js public api tests', function() {
       await expect(privateKey.decrypt('wrong passphrase')).to.eventually.be.rejectedWith('Incorrect key passphrase');
     });
 
-    it('Decrypting key with correct passphrase returns true', async function () {
-      expect(await privateKey.decrypt(passphrase)).to.be.true;
+    it('Decrypting key with correct passphrase does not throw', async function () {
+      await expect(privateKey.decrypt(passphrase)).to.not.be.rejected;
```
Comment (twiss) [inline test/general/openpgp.js:713]:
```suggestion
    it('Can decrypt key with correct passphrase', async function () {
      await privateKey.decrypt(passphrase);
```

File: test/general/openpgp.js [hunk 6]
```diff
@@ -1704,7 +1784,7 @@ module.exports = () => describe('OpenPGP.js public api tests', function() {
       let decryptedPrivateKey;
       beforeEach(async function() {
         if (!decryptedPrivateKey) {
-          expect(await privateKey.decrypt(passphrase)).to.be.true;
+          privateKey.decrypt(passphrase);
```
Comment (twiss) [inline test/general/openpgp.js:1787]:
```suggestion
          await privateKey.decrypt(passphrase);
```

File: test/general/openpgp.js (hunk 5 above)
Comment (larabr) [reply test/general/openpgp.js:713]: I know it's redundant, but in cases such as this one I would not mind being a bit declarative to explicitly mark what we are testing.

**Pull request merged**